packages/core/src/payment/strategies/braintree/is-credit-card-instrument-like.ts

```diff
@@ -0,0 +1,10 @@
+import { OrderPaymentInstrument } from '../../../order';
+import { CreditCardInstrument } from '../../payment';
+
+export default function isCreditCardInstrumentLike(instrument?: OrderPaymentInstrument): instrument is CreditCardInstrument {
+    const paymentData = instrument as CreditCardInstrument;
```

Comment:
Since you are returning `instrument is CreditCardInstrument` here anyway. Think we can avoid this type casting usage by adding a `if`.

Reply:
@bc-peng can you explain what you mean?
do not quite understand